camille/loader.py

```diff
@@ -0,0 +1,95 @@
+import yaml
+import datetime
+import dateutil
+import pytz
+import re
+from functools import partial
+from os import path
+
+from .processors import fft, rolling
+from .inputs import bazefetcher
+
+_processors = {
+    'fft': fft.process,
+    'rolling': rolling.process
+}
+
+_inputs = {
+    'bazefetcher': bazefetcher.load
+}
+
+def run(config):
+    input = config['input']( config )
+    return config['processor']( input )
+
+def _tag(tag):
+    try:
+        return [ re.compile(tag) ]
+    except TypeError:
+        return [ re.compile(t) for t in tag ]
+
+def _to_datetime(time):
+    # Converts to midnight datetime if given as date
+    if not isinstance(time, datetime.datetime):
+        time = datetime.datetime(time.year, time.month, time.day)
+    try:
+        return pytz.utc.localize(time)
+    except ValueError:
+        return time.astimezone(pytz.utc)
+
+def _time(time):
+    if isinstance(time, datetime.date):
+        return _to_datetime(time)
+    msg = "'{0}' is not a valid date/time".format(time)
+    raise ValueError(msg)
+
+def _processor(processor):
+    try:
+        return _processors[processor]
+    except TypeError:
+        function = list(processor.keys())[0]
+        args     = list(processor.values())[0]
+        return partial(_processors[function], **args)
+
+def _input(input):
+    try:
+        return _inputs[input]
+    except TypeError:
+        function = list(input.keys())[0]
+        args     = list(input.values())[0]
+        return partial(_inputs[function], **args)
+
+def load_config(config_path):
+    """Loads a yaml config file
+
+    Config format
+    -------------
+    tag : list of strings
+        Filters for collecting data to be processed. Each string in the list is
+        a regular expression for filtering data to populate a column in the
+        dataframe to be processed.
+    start : date or datetime
+        Collect data from (including) this time.
+    end : date or datetime
+        Collect data before (not including) this time.
+    processor : dict
+        Dict containing a processor name and an optional dict of
+        keyword arguments.
+    input : dict
+        Dict containing an input reader name and an optional dict of
+        keyword arguments.
+    """
+    config_path = path.realpath(config_path)
+
+    with open(config_path, 'r') as f:
```

Comment:
I suggest you check for config_path beeing a yml object already, and if it is, then dont try to read a file, but proceed with the return-parsing-stuff. This enables you to do the actuall loading of YML somewhere else if you want, and it makes it easier to test since you can then construct the  yml in-memory and not write it to a tempfile.